flutter/lib/src/flutter_enricher_event_processor.dart

```diff
@@ -0,0 +1,206 @@
+import 'dart:async';
+import 'dart:ui';
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+import 'package:sentry/sentry.dart';
+
+typedef WidgetBindingGetter = WidgetsBinding Function();
+
+/// Enriches [SentryEvent]s with various kinds of information.
+/// FlutterEnricher only needs to add information which aren't exposed by
+/// the Dart runtime.
+class FlutterEnricherEventProcessor {
+  FlutterEnricherEventProcessor(
+    this._checker,
+    this._getWidgetsBinding,
+    this.hasNativeIntegration,
+  );
+
+  factory FlutterEnricherEventProcessor.simple({
+    required PlatformChecker checker,
+    required bool hasNativeIntegration,
+  }) {
+    return FlutterEnricherEventProcessor(
+      checker,
+      () => WidgetsBinding.instance!,
```

Comment:
is it safe to do `!`? wouldnt cause an issue if its null?

Reply:
At this point it's safe because `WidgetsFlutterBinding.ensureInitialized()` was already called in order to pass Sentry config to native.

I can change it to `?`, it's also probably better.

Reply:
yeah better safe than sorry

Reply:
Just for reference: https://github.com/flutter/flutter/issues/64830